src/ast/semantic_analyser.cpp

```diff
@@ -1406,16 +1406,46 @@ void SemanticAnalyser::visit(If &if_block)
 
 void SemanticAnalyser::visit(Unroll &unroll)
 {
-  if (unroll.var > 20)
+  unroll.expr->accept(*this);
+
+  unroll.var = 0;
+
+  if (auto *integer = dynamic_cast<Integer *>(unroll.expr))
   {
-    error("unroll maximum value is 20", location(0));
+    unroll.var = integer->n;
+  }
+  else if (auto *param = dynamic_cast<PositionalParameter *>(unroll.expr))
+  {
+    if (param->ptype == PositionalParameterType::count)
+    {
+      unroll.var = bpftrace_.num_params();
+    }
+    else
+    {
+      std::string pstr = bpftrace_.get_param(param->n, param->is_in_str);
+      if (is_numeric(pstr))
+        unroll.var = std::stoll(pstr, nullptr, 0);
+      else
+        error("Invalid positonal params: " + pstr, unroll.loc);
+    }
+  }
+  else
+  {
+    out_ << "Unsupported expression" << std::endl;
+    abort();
```

Comment:
```suggestion
error("unroll can only be used..", unroll.loc);
```

Reply:
The intent of `abort()` here is indicating this is a bug if this code is executed (this shouldn't happen due to the grammar). Abort is useful when fuzzing because most fuzzers treat SIGABRT as a bug and report it.

Reply:
oh yes, brainfart